core/src/splitting/RegressionSplittingRule.h

```diff
@@ -32,6 +32,42 @@ class RegressionSplittingRule final: public SplittingRule {
 
   ~RegressionSplittingRule();
 
+  /**
+   * Finds the best split at a given node in the tree.
+   *
+   * Is called repeatedly to build a tree in a DFS or BFS fashion.
+   *
+   * @param data: the data matrix containing all test samples.
+   * @param node: the node id in the tree.
+   * @param possible_split_vars: a vector of valid sample IDs.
+   * @param responses_by_sample: a vector of transformed outcomes.
```

Comment:
Suggested edit: `responses_by_sample: a map from sample ID to response.` I don't think we need to mention that they were transformed, the splitting rule doesn't care about the relabeling step.